osclib/git.py

```diff
@@ -3,18 +3,25 @@
 import subprocess
 
 
-def describe(directory=None):
+def describe(directory=None) -> str:
     if directory:
         cwd = os.getcwd()
         os.chdir(directory)
 
     ret = subprocess.run(
         ['git', 'show', '--no-patch', '--date=short', '--format=%cd.%h'],
-        stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
+        stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+        universal_newlines=True)
 
     if directory:
         os.chdir(cwd)
 
+    # if there are no commits in the current git repo, then the git show call
+    # will fail with status 128 and display the following error:
+    # fatal: your current branch 'main' does not have any commits yet
+    if ret.returncode == 128 and "does not have any commits yet" in ret.stderr:
+        return ""
```

Comment:
This should either throw an exception or return a valid version.

Reply:
It should most certainly not throw an exception, because that will not fix the issue that I am facing.

Reply:
How about this?

Reply:
Which issue are you facing?

I think the cause needs to be understood a bit more to address it properly.

Reply:
What the commit message says: call osc in a git repository with no commits and with this repo installed as mentioned in the readme. You'll then get an exception:
```
❯ mkdir -p /tmp/foobar && pushd /tmp/foobar && git init;
❯ osc st
fatal: your current branch 'main' does not have any commits yet
Failed to load commands from module 'osc.commands.home.osc-staging': Command '['git', 'show', '--no-patch', '--date=short', '--format=%cd.%h']' returned non-zero exit status 128.
fatal: your current branch 'main' does not have any commits yet
Traceback (most recent call last):
  File "/usr/bin/osc", line 33, in <module>
    sys.exit(load_entry_point('osc==1.3.0', 'console_scripts', 'osc')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/osc/babysitter.py", line 234, in main
    sys.exit(run(commandline.OscMainCommand()))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/osc/babysitter.py", line 69, in run
    prg.main(argv)
  File "/usr/lib/python3.11/site-packages/osc/commandline.py", line 542, in main
    cmd.load_legacy_commands()
  File "/usr/lib/python3.11/site-packages/osc/commandline.py", line 521, in load_legacy_commands
    osc_instance = LegacyOsc()
                   ^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/osc/commandline.py", line 960, in __init__
    self._load_plugins()
  File "/usr/lib/python3.11/site-packages/osc/commandline.py", line 10037, in _load_plugins
    spec.loader.exec_module(mod)
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/dan/.osc-plugins/osc-staging.py", line 17, in <module>
    from osclib.accept_command import AcceptCommand
  File "/home/dan/.osc-plugins/osclib/accept_command.py", line 11, in <module>
    from osclib.core import attribute_value_save, attribute_value_load
  File "/home/dan/.osc-plugins/osclib/core.py", line 35, in <module>
    from osclib.conf import Config
  File "/home/dan/.osc-plugins/osclib/conf.py", line 9, in <module>
    from osclib.memoize import memoize
  File "/home/dan/.osc-plugins/osclib/memoize.py", line 5, in <module>
    from osclib.cache_manager import CacheManager
  File "/home/dan/.osc-plugins/osclib/cache_manager.py", line 3, in <module>
    from osclib.common import NAME
  File "/home/dan/.osc-plugins/osclib/common.py", line 16, in <module>
    VERSION = version_calculate()
              ^^^^^^^^^^^^^^^^^^^
  File "/home/dan/.osc-plugins/osclib/common.py", line 9, in version_calculate
    return describe()
           ^^^^^^^^^^
  File "/home/dan/.osc-plugins/osclib/git.py", line 18, in describe
    ret.check_returncode()
  File "/usr/lib64/python3.11/subprocess.py", line 502, in check_returncode
    raise CalledProcessError(self.returncode, self.args, self.stdout,
subprocess.CalledProcessError: Command '['git', 'show', '--no-patch', '--date=short', '--format=%cd.%h']' returned non-zero exit status 128.
```
So I am trying to catch the case when there are no commits on the current branch yet.

Reply:
Looks like that is meant to get the version of `openSUSE-release-tools` itself, but for some reason it calls `git` in the CWD, not the release-tools dir...

```
fvogt@linux-e202:~/osc-plugin-factory> osc staging --version
20230731.178896cc1fda
fvogt@linux-e202:~/osc-plugin-factory> cd
fvogt@linux-e202:~> osc staging --version
0.0.0-dev
```

Reply:
I've changed it to `cd` into the opensuse-release-tools dir.